### PR TITLE
Refactor rules definition functionality - add rules module to read rule definition files

### PR DESCRIPTION
**Pull request by awang-parasoft**

### CODE REVIEW

File: src/main/java/com/parasoft/findings/utils/common/util/XMLUtil.java [hunk 1]
```diff
@@ -63,34 +66,70 @@ public static SAXParserFactory getSaxParserFactory() {
         return _saxParserFactory;
     }
 
+    static boolean shouldDisallowDTD()
```
Comment (rwang-parasoft) [inline src/main/java/com/parasoft/findings/utils/common/util/XMLUtil.java:69]: This method always returns true(we don't need to use system property), consider removing it and related code.

File: src/main/java/com/parasoft/findings/utils/common/util/LocaleUtil.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    private LocaleUtil() {}
+
+    /**
+     * Returns the default locale to use. This setting can be overridden by using the system
+     * properties <code>JTEST_LOCALE_LANGUAGE</code> and <code>JTEST_LOCALE_COUNTRY</code>.
+     *
+     * @return the default locale to use
+     */
+    public static synchronized Locale getLocale()
```
Comment (rwang-parasoft) [inline src/main/java/com/parasoft/findings/utils/common/util/LocaleUtil.java:39]: This method always returns default locale(we don't need to use system property), consider removing it and call `Locale.getDefault()` where calls this method.

File: src/main/java/com/parasoft/findings/utils/rules/RuleParserUtil.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    public static void saxParse(URL file, DefaultHandler handler, RuleParsingOptions options)
+        throws IOException
+    {
+        InputStream in = null;
+        try {
+            in = file.openStream();
+            saxParse(in, handler, options);
+        } catch (FileNotFoundException e) {
+            Logger.getLogger().debug("File not found under given url " + file); //$NON-NLS-1$
```
Comment (rwang-parasoft) [inline src/main/java/com/parasoft/findings/utils/rules/RuleParserUtil.java:45]: I think it is better to change debug to warn.

File: src/main/java/com/parasoft/findings/utils/rules/CategoryDescription.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * @param sCategoryId
+     * @param sDescription
+     * @param separator
+     *
+     * @pre sCategoryId != null
+     * @pre sDescription != null
+     */
+    public CategoryDescription(String sCategoryId, String sDescription, char separator)
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/findings/utils/rules/CategoryDescription.java:33]: Seems `separator` and `sDescription` are not used

File: src/main/java/com/parasoft/findings/utils/rules/RuleDescription.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+import com.parasoft.findings.utils.common.util.StringUtil;
+
+public class RuleDescription
+{
+    private String _sRuleId = null;
+    private String _sHeader = null;
+
+    private int _severity = -1;
+    private RuleScope _scope = RuleScope.line;
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/findings/utils/rules/RuleDescription.java:30]: Seems `_scope`, `_bQuickfix` and `_parameters` are not used

File: src/main/java/com/parasoft/findings/utils/rules/RuleDescriptionParser.java [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class RuleDescriptionParser
+{
+    private static final char _separator = '.';
+
+    private final Locale _locale;
+    private final List<RuleDescription> _rules = new ArrayList<RuleDescription>();
+    private final List<CategoryDescription> _categories = new ArrayList<CategoryDescription>();
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/findings/utils/rules/RuleDescriptionParser.java:33]: Seems `_categories` is updated but never used.

**Pull request merged**